### PR TITLE
Force bytestrings to string in tomcat module

### PR DESCRIPTION
**Pull request by sabtom**
### What does this PR do?
This PR fixes the issue #56269 
It converts the byte strings into strings before using it elsewhere in the module.

### What issues does this PR fix or reference?
#56269 

### Tests written?
No, current tests should still be adequate/behave the same

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by sabtom** [comment]
FYI, just checked the linter fails, because some dependencies arent installed:
``` bash
12:11:44  Command pylint --rcfile=.pylintrc --disable=I setup.py noxfile.py salt/ failed with exit code 2
12:11:44  ************* Module salt.returners.django_return
12:11:44  salt/returners/django_return.py:42: [E0611(no-name-in-module), ] No name 'dispatch' in module 'django'
```

**Comment by sabtom** [comment]
Also checked why the macosxmojave failed, but that's also seems to be because of missing dependencies:
```
13:39:12            -> unit.beacons.test_twilio_txt_msg.TwilioMsgTxtBeaconTestCase.test_receive_message                                                                                                                                                                                      ->  twilio.rest is not available
13:39:12            -> unit.beacons.test_twilio_txt_msg.TwilioMsgTxtBeaconTestCase.test_validate_dictionary_config                                                                                                                                                                           ->  twilio.rest is not available
13:39:12            -> unit.beacons.test_twilio_txt_msg.TwilioMsgTxtBeaconTestCase.test_validate_empty_config                                                                                                                                                                                ->  twilio.rest is not available
13:39:12            -> unit.beacons.test_twilio_txt_msg.TwilioMsgTxtBeaconTestCase.test_validate_missing_config_item                                                                                                                                                                         ->  twilio.rest is not available
```

**Review by twangboy — DISMISSED** [review]

**Comment by twangboy** [comment]
@sabtom Tests need to be written to address this issue so it doesn't regress. Since it wasn't caught by current tests, we need to make sure we would catch this in the future.

**Comment by sabtom** [comment]
@twangboy Apparently there didn't exists a unittest for module/tomcat yet, so added it.  
Don't really have time to spend to test all methods, but have added a test that will fail when a bytestring is returned by the _wget method of the tomcat module.

**Comment by sabtom** [comment]
Hmm all tests seem to fail on the zcbuildout module, but haven't touched that one.
```
09:40:12          -----------------------------------------------------------------------------------------------------------------------
09:40:12          --------  Tests with Errors  ------------------------------------------------------------------------------------------
09:40:12            -> setUpClass (unit.modules.test_zcbuildout.BuildoutOnlineTestCase)  ................................................
09:40:12         Traceback (most recent call last):
09:40:12           File "/tmp/kitchen/testing/tests/unit/modules/test_zcbuildout.py", line 353, in setUpClass
09:40:12             super(BuildoutOnlineTestCase, cls).setUpClass()
09:40:12           File "/tmp/kitchen/testing/tests/unit/modules/test_zcbuildout.py", line 91, in setUpClass
09:40:12             download_to(url, dest)
09:40:12           File "/tmp/kitchen/testing/tests/unit/modules/test_zcbuildout.py", line 61, in download_to
09:40:12             fic.write(urlopen(url, timeout=10).read())
09:40:12           File "/usr/lib/python3.6/urllib/request.py", line 223, in urlopen
09:40:12             return opener.open(url, data, timeout)
09:40:12         AttributeError: 'bool' object has no attribute 'open'
09:40:12            .....................................................................................................................
09:40:12            -> setUpClass (unit.modules.test_zcbuildout.BuildoutTestCase)  ......................................................
09:40:12         Traceback (most recent call last):
09:40:12           File "/tmp/kitchen/testing/tests/unit/modules/test_zcbuildout.py", line 91, in setUpClass
09:40:12             download_to(url, dest)
09:40:12           File "/tmp/kitchen/testing/tests/unit/modules/test_zcbuildout.py", line 61, in download_to
09:40:12             fic.write(urlopen(url, timeout=10).read())
09:40:12           File "/usr/lib/python3.6/urllib/request.py", line 223, in urlopen
09:40:12             return opener.open(url, data, timeout)
09:40:12         AttributeError: 'bool' object has no attribute 'open'
09:40:12            .....................................................................................................................
09:40:12            -> setUpClass (unit.states.test_zcbuildout.BuildoutTestCase)  .......................................................
09:40:12         Traceback (most recent call last):
09:40:12           File "/tmp/kitchen/testing/tests/unit/modules/test_zcbuildout.py", line 91, in setUpClass
09:40:12             download_to(url, dest)
09:40:12           File "/tmp/kitchen/testing/tests/unit/modules/test_zcbuildout.py", line 61, in download_to
09:40:12             fic.write(urlopen(url, timeout=10).read())
09:40:12           File "/usr/lib/python3.6/urllib/request.py", line 223, in urlopen
09:40:12             return opener.open(url, data, timeout)
09:40:12         AttributeError: 'bool' object has no attribute 'open'
09:40:12            .....................................................................................................................
09:40:12          -----------------------------------------------------------------------------------------------------------------------
09:40:12         ========================================================================================================================
09:40:12         FAILED (total=10532, skipped=1430, passed=9099, failures=0, errors=3) 
09:40:12         ================================================  Overall Tests Report  ================================================
```

File: salt/modules/tomcat.py [hunk 1]
```diff
@@ -259,6 +259,13 @@ def _wget(cmd, opts=None, url='http://localhost:8080/manager', timeout=180):
         except Exception:  # pylint: disable=broad-except
             ret['msg'] = 'Failed to create HTTP request'
 
+    # Force all byte strings to utf-8 strings, for python >= 3.4
+    for key, item in enumerate(ret['msg']):
+        try:
+            ret['msg'][key] = item.decode('utf-8')
```
Comment (DmitryKuzmenko) [inline salt/modules/tomcat.py:265]: Could you please use `salt.utils.stringutils.to_unicode` instead?

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]

**Review by DmitryKuzmenko — DISMISSED** [review]

**Comment by DmitryKuzmenko** [comment]
@sabtom could you please resolve the merge conflicts?

**Comment by dwoz** [comment]
@sabtom Maybe try re-basing this on top of the current master and running black as a part of the pre-commit. Hopefully that'll resolve the weird test failures.

File: tests/unit/modules/test_tomcat.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    def test_tomcat_wget_no_bytestring(self):
+        responses = {
+            "string": StringIO("Best response ever\r\nAnd you know it!"),
+            "bytes": BytesIO(b"Best response ever\r\nAnd you know it!"),
+        }
+
+        string_mock = MagicMock(return_value=responses["string"])
+        bytes_mock = MagicMock(return_value=responses["bytes"])
+        with patch("salt.modules.tomcat._auth", MagicMock(return_value=True)):
```
Comment (DmitryKuzmenko) [inline tests/unit/modules/test_tomcat.py:35]: Here is what is breaking the zcbuildout tests.
The `_auth` function in tomcat module either returns `False` that makes the underlying `_wget` to return an error or returns an `opener` that is being directly installed as a default opener to urllib that breaks urllib for further calls.

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]
Please update the test to not return `True` from the `tomcat._auth` module function.

**Comment by DmitryKuzmenko** [comment]
@sabtom thank you! Now it should be good!

**Review by dwoz — APPROVED** [review]

**Pull request merged**